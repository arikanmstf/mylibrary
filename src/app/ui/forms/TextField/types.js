/**
 * Component Types By => create-module script
 * @version 1.0.0
 *
 */

// @flow
export type TextFieldProps = {
  label?: string,
  name: string,
  type?: string,
  style?: Object,
};

export type ReduxFieldProps = {
  meta: {
    touched?: ?boolean,
    error?: ?string,
  },
  input: {
    name: string,
    value: *,
    onChange: Function,
  },
}
