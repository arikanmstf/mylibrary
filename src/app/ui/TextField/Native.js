/**
 * Native Component Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow

const TextField = (): ?Node => {
  return null;
};

export default TextField;
