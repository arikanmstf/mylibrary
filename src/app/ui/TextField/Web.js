/**
 * Web Component Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow
import React from 'react';
import WebTextField from '@material-ui/core/TextField';

import type { Node } from 'react';
import type { TextFieldProps } from './types';

const TextField = (props: TextFieldProps): Node => {
  return <WebTextField {...props} />;
};

export default TextField;
