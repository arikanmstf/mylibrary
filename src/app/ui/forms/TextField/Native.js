/**
 * Native Component Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow
import React from 'react';
import { StyleSheet, TextInput } from 'react-native';
import { Field } from 'redux-form/immutable';
import type { Node } from 'react';

import defaultStyle from './style.json';

import type { TextFieldProps, ReduxFieldProps } from './types';

const renderTextField = ({ input, ...other }: ReduxFieldProps) => (
  <TextInput
    onChangeText={input.onChange}
    {...input}
    {...other}
  />
);

const TextField = (props: TextFieldProps): ?Node => {
  const {
    label,
    style,
    type,
    ...other
  } = props;

  const mergedStyles = StyleSheet.create({
    container: {
      ...defaultStyle,
      ...style,
    },
  });
  const isPassword = type === 'password';

  return (
    <Field
      component={renderTextField}
      placeholder={label}
      style={mergedStyles.container}
      {...other}
      secureTextEntry={isPassword}
    />
  );
};

export default TextField;
