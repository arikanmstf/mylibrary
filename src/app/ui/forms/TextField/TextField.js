/**
 * Web Component Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow
import React from 'react';
import WebTextField from '@material-ui/core/TextField';
import { Field } from 'redux-form/immutable';

import type { Node } from 'react';
import type { TextFieldProps } from './types';
import type { ReduxFieldProps } from '../types';

import defaultStyle from './style.json';

const renderTextField = ({
  input, meta: { touched, error }, label, ...other
}: ReduxFieldProps) => (
  <WebTextField
    error={!!(touched && error)}
    label={(touched && error) ? `${label}, ${error}` : label}
    {...input}
    {...other}
  />
);

const TextField = (props: TextFieldProps): Node => {
  const {
    style,
    name,
    autoComplete,
    ...other
  } = props;

  const mergedStyle = {
    ...defaultStyle,
    width: '100%',
    ...style,
  };

  return (
    <Field
      component={renderTextField}
      name={name}
      style={mergedStyle}
      autoComplete={autoComplete || 'off'}
      {...other}
    />
  );
};

export default TextField;
