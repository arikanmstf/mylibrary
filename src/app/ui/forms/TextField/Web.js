/**
 * Web Component Template By => create-module script
 * @version 1.0.0
 *
 */

// @flow
import React from 'react';
import WebTextField from '@material-ui/core/TextField';
import { Field } from 'redux-form/immutable';

import type { Node } from 'react';
import type { TextFieldProps, ReduxFieldProps } from './types';
import defaultStyle from './style.json';

const renderTextField = ({
  input, meta: { touched, error }, ...other
}: ReduxFieldProps) => (
  <WebTextField
    errorText={touched && error}
    {...input}
    {...other}
  />
);

const TextField = (props: TextFieldProps): Node => {
  const { style, name, ...other } = props;
  const mergedStyle = {
    ...defaultStyle,
    width: '100%',
    ...style,
  };

  return (
    <Field
      component={renderTextField}
      name={name}
      style={mergedStyle}
      {...other}
    />
  );
};

export default TextField;
