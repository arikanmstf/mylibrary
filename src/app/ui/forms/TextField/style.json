{
  "backgroundColor": "#555",
  "minHeight": 50
}
