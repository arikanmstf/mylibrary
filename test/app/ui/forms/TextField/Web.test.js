/**
 * Web Component Test Template By => create-module script
 * @version 1.0.0
 *
 */

import React from 'react';
import { shallow } from 'enzyme';
import TextField from 'ui/forms/TextField/Web';

describe('test/app/ui/forms/TextField/Web.test.js', () => {
  it('Render', () => {
    const props = {};
    const wrapper = shallow(<TextField {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
