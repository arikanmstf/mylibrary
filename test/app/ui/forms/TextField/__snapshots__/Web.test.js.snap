// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test/app/ui/forms/TextField/Web.test.js Render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextField />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "component": [Function],
      "name": undefined,
      "style": Object {
        "maxHeight": 50,
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "component": [Function],
        "name": undefined,
        "style": Object {
          "maxHeight": 50,
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
